lld/ELF/SyntheticSections.cpp

```diff
@@ -2753,6 +2753,21 @@ RelroPaddingSection::RelroPaddingSection(Ctx &ctx)
     : SyntheticSection(ctx, ".relro_padding", SHT_NOBITS, SHF_ALLOC | SHF_WRITE,
                        1) {}
 
+ShufflePaddingSection::ShufflePaddingSection(Ctx &ctx, uint64_t size,
+                                             OutputSection *parent)
+    : SyntheticSection(ctx, ".shuffle_padding", SHF_ALLOC, SHT_PROGBITS, 1),
```

Comment:
The new sections are added after the output section flags are resolved, so I don't think the values here matter. I just picked values arbitrarily. The new tests I added show that the correct section flags are preserved.

Reply:
OK I see. Could possibly use `parent->flags`, and `parent->type` instead. I don't think it will make a difference at the moment, but may be a bit less surprising see.